File: js/importing-assets/CreateAsset.js
// DocSection: importing_assets_create_asset
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

// Uses the file reference object obtained in step 1
client.upsertAsset()
  .withData(
    {
      fileReference: {
        id: '8660e19c-7bbd-48a3-bb51-721934c7756c',
        type: 'internal'
      },
      title: "Brno Cafe",
      assetExternalId: 'brno-cafe-image',
      descriptions: [
        {
          language: {
            codename: 'en-US'
          },
          description: 'Cafe in Brno'
        },
        {
          language: {
            codename: 'es-ES'
          },
          description: 'Café en Brno'
        }
      ]
    }
  )
  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
    (error) => {
      console.log(error);
    });
// EndDocSection

File: net/import-assets/CreateAsset.cs
// DocSection: importing_assets_create_asset
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
// Using Management API v1
AssetDescription englishDescription = new AssetDescription
{
    Description = "Cafe in Brno",
    Language = LanguageIdentifier.ByCodename("en-US")
};
AssetDescription spanishDescription = new AssetDescription
{
    Description = "Café en Brno",
    Language = LanguageIdentifier.ByCodename("es-ES")
};
IEnumerable<AssetDescription> descriptions = new [] { englishDescription, spanishDescription };

string title = "Brno Cafe";

// Defines the asset to upsert
AssetUpsertModel asset = new AssetUpsertModel
{
    // Uses the file reference object obtained in step 1
    FileReference = fileResult,
    Descriptions = descriptions,
    Title = title
};

string assetExternalId = "brno-cafe-image";

// Upserts an asset by external ID
AssetModel response = await client.UpsertAssetByExternalIdAsync(externalId, asset);
// EndDocSection